manageiq-operator/pkg/helpers/miq-components/postgresql.go

```diff
@@ -9,6 +12,46 @@ import (
 	intstr "k8s.io/apimachinery/pkg/util/intstr"
 )
 
+func DefaultPostgresqlSecret(cr *miqv1alpha1.Manageiq) *corev1.Secret {
+	labels := map[string]string{
+		"app": cr.Spec.AppName,
+	}
+	secret := map[string]string{
+		"dbname":   "vmdb_production",
+		"username": "root",
+		"password": generateDatabasePassword(),
+		"hostname": "postgresql",
+		"port":     "5432",
+	}
+
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      PostgresqlSecretName(cr),
+			Namespace: cr.ObjectMeta.Namespace,
+			Labels:    labels,
+		},
+		StringData: secret,
+	}
+}
+
+func PostgresqlSecretName(cr *miqv1alpha1.Manageiq) string {
+	secretName := "postgresql-secrets"
+	if cr.Spec.DatabaseSecret != "" {
+		secretName = cr.Spec.DatabaseSecret
+	}
+
+	return secretName
+}
+
+func generateDatabasePassword() string {
+	buf := make([]byte, 8)
+	_, err := rand.Read(buf)
+	if err != nil {
+		panic(err) // out of randomness, should never happen
+	}
+	return hex.EncodeToString(buf)
+}
```

Comment:
This feels like an almost copy/paste of the other secrets method [here](https://github.com/ManageIQ/manageiq-pods/blob/1ebf278f5adb1113a3cd15d127b073a8d50198cd/manageiq-operator/pkg/helpers/miq-components/app-secret.go#L32-L43).  I'm wondering if it's better to have a utility module where the baseline logic for generating passwords lives.  Perhaps taking a parameter on the size of the buffer?

Reply:
Yeah, I was debating extracting this, but I couldn't find a good place for it just yet.

I could make a separate file in the `miqtools` package for the utility functions ... was also debating turning the `tlstools` package into a more general secret tools thing ...

I ended up punting and just opening this as is.